string label
---
bool success
string path